File: android/src/main/kotlin/com/example/konnek_flutter/response/GetConfigResponseModel.kt
package com.example.konnek_flutter.response

import kotlinx.parcelize.Parcelize
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.RawValue

@Parcelize
data class GetConfigResponseModel(

	@field:SerializedName("data")
	val data: DataGetConfig? = null,

	@field:SerializedName("meta")
	val meta: MetaGetConfig? = null
) : Parcelable

@Parcelize
data class DataGetConfig(

	@field:SerializedName("preview")
	val preview: String? = null,

	@field:SerializedName("background_status")
	val backgroundStatus: String? = null,

	@field:SerializedName("ios_icon")
	val widgetIcon: String? = null,

	@field:SerializedName("company_id")
	val companyId: String? = null,

	@field:SerializedName("header_text_color")
	val headerTextColor: String? = null,

	@field:SerializedName("text_button_color")
	val textButtonColor: String? = null,

	@field:SerializedName("avatar_image")
	val avatarImage: String? = null,

	@field:SerializedName("text_button")
	val textButton: String? = null,

	@field:SerializedName("header_background_color")
	val headerBackgroundColor: String? = null,

	@field:SerializedName("button_color")
	val buttonColor: String? = null,

	@field:SerializedName("greeting_message")
	val greetingMessage: String? = null,

	@field:SerializedName("background")
	val background: String? = null,

	@field:SerializedName("avatar_name")
	val avatarName: String? = null,

	@field:SerializedName("status")
	val status: Boolean? = null,

	@field:SerializedName("text_status")
	val textStatus: Boolean? = null
) : Parcelable

@Parcelize
data class MetaGetConfig(

	@field:SerializedName("log_id")
	val logId: String? = null,

	@field:SerializedName("code")
	val code: Int? = null,

	@field:SerializedName("message")
	val message: String? = null,

	@field:SerializedName("errors")
	val errors: @RawValue Any? = null,

	@field:SerializedName("status")
	val status: Boolean? = null
) : Parcelable

File: ios/Classes/response/GetConfigResponseModel.swift
//import Foundation
//
//struct GetConfigResponseModel: Codable {
//    let data: DataGetConfig?
//    let meta: MetaGetConfig?
//}
//
//struct DataGetConfig: Codable {
//    let preview: String?
//    let backgroundStatus: String?
//    let widgetIcon: String?
//    let companyId: String?
//    let headerTextColor: String?
//    let textButtonColor: String?
//    let avatarImage: String?
//    let textButton: String?
//    let headerBackgroundColor: String?
//    let buttonColor: String?
//    let greetingMessage: String?
//    let background: String?
//    let avatarName: String?
//    let status: Bool?
//
//    enum CodingKeys: String, CodingKey {
//        case preview
//        case backgroundStatus = "background_status"
//        case widgetIcon = "widget_icon"
//        case companyId = "company_id"
//        case headerTextColor = "header_text_color"
//        case textButtonColor = "text_button_color"
//        case avatarImage = "avatar_image"
//        case textButton = "text_button"
//        case headerBackgroundColor = "header_background_color"
//        case buttonColor = "button_color"
//        case greetingMessage = "greeting_message"
//        case background
//        case avatarName = "avatar_name"
//        case status
//    }
//}
//
//struct MetaGetConfig: Codable {
//    let logId: String?
//    let code: Int?
//    let message: String?
//    let errors: AnyCodable?  // We'll handle dynamic typing here
//    let status: Bool?
//
//    enum CodingKeys: String, CodingKey {
//        case logId = "log_id"
//        case code
//        case message
//        case errors
//        case status
//    }
//}
